src/Frontend/Page/Idea.hs

```diff
@@ -270,13 +270,12 @@ instance FormPage EditIdea where
         <*> ("idea-category" .: makeFormSelectCategory (idea ^. ideaCategory))
         <*> pure (idea ^. ideaLocation)
 
-    -- FIXME: factor out code common with CreateIdea.
-    -- FIXME: category choice should look like in CreateIdea.
-    formPage = createOrEditPage True
+    formPage v form p@(EditIdea idea) = createOrEditPage True (idea ^. ideaLocation) v form p
 
 createOrEditPage :: (Monad m, Typeable page, Page page) =>
-    Bool -> View (HtmlT m ()) -> (HtmlT m () -> HtmlT m ()) -> page -> HtmlT m ()
-createOrEditPage showDeleteButton v form p = semanticDiv p $ do
+    Bool -> IdeaLocation ->
+    View (HtmlT m ()) -> (HtmlT m () -> HtmlT m ()) -> page -> HtmlT m ()
+createOrEditPage showDeleteButton cancelUrl v form p = semanticDiv p $ do
```

Comment:
Actually, i changed my mind.  `showDeleteButton` is set by one of the two callers of this, which are the two `ToHtml` instances of `EditIdea`, `CreateIdea` resp.

So this is not really about the session context, but about which of two similar forms we happen to render.

Does that make sense?